### PR TITLE
Python Frontend AC will call C++ implementation of AC

### PR DESCRIPTION
**Pull request by shreyajain17**
When we set `use_tensorflow=True` in `_advanced_parameters`, we basically use the C++ implementation of AC now. The methods create, predict and evaluate work for now. I still have to get the other stuff working. 

Please only look at `_activity_classifier.py` only. Working on the other files in a different PR (will rebase before merging).

### CODE REVIEW

File: src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import time as _time
 import six as _six
 
+import turicreate as _turicreate
```
Comment (nickjong) [inline src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py]: Why do we need this at the top level?

File: src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py [hunk 2]
```diff
@@ -165,7 +167,36 @@ def create(dataset, session_id, target, features=None, prediction_window=100,
...
+
+        name = 'activity_classifier'
+
+        # Imports tensorflow
+        import turicreate.toolkits.libtctensorflow
+        from ._tf_model_architecture import ActivityTensorFlowModel
+
+
+        model = _turicreate.extensions.activity_classifier()
+        options = {}
```
Comment (nickjong) [inline src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py]: What about `features`?

File: src/ml/neural_net/tf_compute_context.hpp [hunk 3]
```diff
@@ -47,6 +67,22 @@ class tf_compute_context: public compute_context {
   
 };
 
+class tf_model_backend : public model_backend {
```
Comment (nickjong) [inline src/ml/neural_net/tf_compute_context.hpp]: I don't think this even needs to be in the header file. If we move this to the top of the `.cpp` file, then all the pybind headers can go there too

File: src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py (hunk 2 above)
Comment (shreyajain17) [reply src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py]: passing it in the dataset, right?

File: src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py (hunk 1 above)
Comment (shreyajain17) [reply src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py]: Won't I need it to call extensions?

File: src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py (hunk 2 above)
Comment (shreyajain17) [reply src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py]: The options dictionary I made to sent the arguments needed for init_options. There aren't any features. I am sending the features when I call train right?

File: src/ml/neural_net/tf_compute_context.hpp (hunk 3 above)
Comment (shreyajain17) [reply src/ml/neural_net/tf_compute_context.hpp]: Doing it in the other PR.

**Pull request closed without merging**